examples/sync-server/index.js

```diff
@@ -0,0 +1,81 @@
+// @ts-check
+import fs from "fs"
+import express from "express"
+import { WebSocketServer } from "ws"
+import { Repo } from "@automerge/automerge-repo"
+import { NodeWSServerAdapter } from "@automerge/automerge-repo-network-websocket"
+import { NodeFSStorageAdapter } from "@automerge/automerge-repo-storage-nodefs"
+import os from "os"
+
+export class Server {
+  /** @type WebSocketServer */
+  #socket
+
+  /** @type ReturnType<import("express").Express["listen"]> */
+  #server
+
+  /** @type {((value: any) => void)[]} */
+  #readyResolvers = []
+
+  #isReady = false
+
+  constructor() {
+    const dir = ".amrg"
+    if (!fs.existsSync(dir)) {
+      fs.mkdirSync(dir)
+    }
+
+    var hostname = os.hostname()
+
+    this.#socket = new WebSocketServer({ noServer: true })
+
+    const PORT =
+      process.env.PORT !== undefined ? parseInt(process.env.PORT) : 3030
+    const app = express()
+    app.use(express.static("public"))
+
+    const config = {
+      network: [new NodeWSServerAdapter(this.#socket)],
+      storage: new NodeFSStorageAdapter(dir),
+      /** @ts-ignore @type {(import("automerge-repo").PeerId)}  */
+      peerId: `storage-server-${hostname}`,
+      // Since this is a server, we don't share generously — meaning we only sync documents they already
+      // know about and can ask for by ID.
+      sharePolicy: async () => false,
+    }
+    const serverRepo = new Repo(config)
+
+    app.get("/", (req, res) => {
+      res.send(`👍 @automerge/automerge-repo-sync-server is running`)
```

Comment:
Not strictly necessary, but might be worth renaming this to match the example package name too